src/service/resources-impl.js

```diff
@@ -407,13 +410,29 @@ export class Resources {
    * @private
    */
   buildReadyResources_() {
+    // Avoid cases where elements add more elements inside of them
```

Comment:
Should we go ahead and modify the code that puts the element into the queue if it has already been built? E.g.

```
this.pendingBuildResources_.push(resource);
this.buildReadyResources_();
```

Becomes 
```
if (resource.isBuilt())  {
  this.pendingBuildResources_.push(resource);
  this.buildReadyResources_();
}
```

I know it's pretty benign to try to build again, but why do more work if we don't have to.